### PR TITLE
OSASINFRA-3557: Clean Machine Network variable definitions

### PR DESCRIPTION
**Pull request by MiguelCarpio**
There's a good opportunity to clean things up - MachineNet

### CODE REVIEW

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.16-e2e-openstack-network-perf

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.16-e2e-openstack-network-perf

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.16-e2e-openstack-network-perf

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.16-e2e-openstack-network-perf

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.16-e2e-openstack-network-perf

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.16-e2e-openstack-network-perf

File: ci-operator/step-registry/openstack/provision/securitygroups/openstack-provision-securitygroups-commands.sh [hunk 1]
```diff
@@ -29,7 +29,11 @@ if [[ -n "$ADDITIONAL_SECURITY_GROUP_RULES" ]]; then
 			case $service in
 				netperf)
 					echo "Adding ${service} rule to security group ${sg_id}" 
-					openstack security group rule create "$sg_id" --protocol tcp --dst-port 12865:12865 --remote-ip 0.0.0.0/0
+					openstack security group rule create "$sg_id" --protocol tcp --dst-port 12865:12865 --remote-ip 0.0.0.0/0 --description netperf
+					openstack security group rule create "$sg_id" --protocol tcp --dst-port 22865:22865 --remote-ip 0.0.0.0/0 --description iperf3
+					openstack security group rule create "$sg_id" --protocol tcp --dst-port 30000:30000 --remote-ip 0.0.0.0/0 --description uperf
+					openstack security group rule create "$sg_id" --protocol tcp --dst-port 32000:47000 --remote-ip 0.0.0.0/0 --description netserver-tcp
```
Comment (mandre) [inline ci-operator/step-registry/openstack/provision/securitygroups/openstack-provision-securitygroups-commands.sh:35]: Would you have links for these port ranges? they're pretty large, is there a way to make them smaller?

File: ci-operator/step-registry/openstack/conf/clouds/openstack-conf-clouds-commands.sh [hunk 2]
```diff
@@ -40,7 +40,7 @@ new_application_credentials() {
 			"${*:---restricted}" \
 			--expiration "$(date -d "$APPLICATION_CREDENTIALS_EXPIRATION" +%Y-%m-%dT%H:%M:%S)" \
 			--description "PROW_CLUSTER_NAME=${CLUSTER_NAME} PROW_JOB_ID=${PROW_JOB_ID}" \
-			--format json --column id --column secret \
+			--format json --column ID --column Secret \
```
Comment (mandre) [inline ci-operator/step-registry/openstack/conf/clouds/openstack-conf-clouds-commands.sh:43]: Is this change needed? It feels out of place.

File: ci-operator/step-registry/openstack/provision/securitygroups/openstack-provision-securitygroups-commands.sh (hunk 1 above)
Comment (mandre) [reply ci-operator/step-registry/openstack/provision/securitygroups/openstack-provision-securitygroups-commands.sh:35]: Also, if we know the IP range from where this connection is supposed to initiate, we can restrict the source IP, and stop using 0.0.0.0/0.

File: ci-operator/step-registry/openstack/conf/clouds/openstack-conf-clouds-commands.sh (hunk 2 above)
Comment (MiguelCarpio) [reply ci-operator/step-registry/openstack/conf/clouds/openstack-conf-clouds-commands.sh:43]: I hope no, just testing and issue in openstack-conf-clouds step

File: ci-operator/step-registry/openstack/provision/securitygroups/openstack-provision-securitygroups-commands.sh (hunk 1 above)
Comment (MiguelCarpio) [reply ci-operator/step-registry/openstack/provision/securitygroups/openstack-provision-securitygroups-commands.sh:35]: It works if I set the `machineNetwork` to `netserver-tcp` and `netserver-udp` rules. But at this point `openstack-provision-securitygroups` we don't know what is going to be the `machineNetwork`. By default, if it runs the `minimal` type, it will be the 10.0.0.0/16 due to openshift-install will create a machine subnet with that CIDR.

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.16-e2e-openstack-network-perf

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.16-e2e-openstack-network-perf

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.16-e2e-openstack-network-perf

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.16-e2e-openstack-network-perf

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.16-e2e-openstack-network-perf

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.16-e2e-openstack-network-perf

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.16-e2e-openstack-network-perf

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.16-e2e-openstack-network-perf

File: ci-operator/step-registry/openstack/provision/securitygroups/openstack-provision-securitygroups-ref.yaml [hunk 3]
```diff
@@ -14,5 +14,25 @@ ref:
...
+    - name: CONFIG_TYPE
+      documentation: |
+        The type of config for the environment to deploy.
+
+        * 'minimal' - Configure the install-config with the minimal options to make it work on
+                      our tests, which requires the usage of floating IPs. This config type doesn't
+                      require a proxy server to be deployed.
+        * 'proxy' - Configure the install-config to use a pre-created restricted network (BYON) and
+                   a Cluster Proxy.
+    - name: MACHINES_SUBNET_v4_RANGE
```
Comment (pierreprinetti) [inline ci-operator/step-registry/openstack/provision/securitygroups/openstack-provision-securitygroups-ref.yaml:26]:
```suggestion
    - name: MACHINES_SUBNET_V4_RANGE
```

File: ci-operator/step-registry/openstack/provision/securitygroups/openstack-provision-securitygroups-ref.yaml [hunk 4]
```diff
@@ -14,5 +14,25 @@ ref:
...
+      default: "192.168.25.0/24"
+      documentation: |-
+        CIDR of the IPv4 Machines Subnet
+    - name: MACHINES_SUBNET_v6_RANGE
```
Comment (pierreprinetti) [inline ci-operator/step-registry/openstack/provision/securitygroups/openstack-provision-securitygroups-ref.yaml:30]:
```suggestion
    - name: MACHINES_SUBNET_V6_RANGE
```

**Review by pierreprinetti — COMMENTED** [review]
Isn't the machines subnet programmatically retrievable?

File: ci-operator/step-registry/openstack/provision/securitygroups/openstack-provision-securitygroups-ref.yaml (hunk 4 above)
Comment (MiguelCarpio) [reply ci-operator/step-registry/openstack/provision/securitygroups/openstack-provision-securitygroups-ref.yaml:30]: I have the same observation, but to use the same v6 and v4 machine subnet range I need to use the same env variable defined [here](https://github.com/openshift/release/blob/master/ci-operator/step-registry/openstack/conf/installconfig/openstack-conf-installconfig-ref.yaml) . However, I'll change the strategy and get only the v4 machine subnet for minimal config type that I need in this PR.

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.16-e2e-openstack-network-perf

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.16-e2e-openstack-network-perf

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.17-e2e-openstack-dualstack-upi

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.17-e2e-openstack-ovn-password

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.16-e2e-openstack-network-perf

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.17-e2e-openstack-dualstack-upi

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.17-e2e-openstack-dualstack-upi

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.17-e2e-openstack-dualstack-upi

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.17-e2e-openstack-dualstack

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.17-e2e-openstack-dualstack

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.13-e2e-openstack-proxy

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.13-e2e-openstack-proxy

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.17-e2e-openstack-dualstack

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.16-e2e-openstack-network-perf

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.17-e2e-openstack-dualstack-upi

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.16-e2e-openstack-dualstack-v6primary

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.16-e2e-openstack-dualstack-v6primary

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.15-e2e-openstack-additional-ipv6-network

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.16-e2e-openstack-dualstack-v6primary

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.15-e2e-openstack-additional-ipv6-network

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.15-e2e-openstack-additional-ipv6-network

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.15-e2e-openstack-additional-ipv6-network

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.15-e2e-openstack-nfv-intel

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.15-e2e-openstack-nfv-intel

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.14-e2e-openstack-nfv-mellanox

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.14-e2e-openstack-nfv-mellanox

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.15-e2e-openstack-nfv-intel

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.14-e2e-openstack-nfv-mellanox

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.15-e2e-openstack-nfv-intel

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.14-e2e-openstack-nfv-mellanox

File: ci-operator/step-registry/openstack/provision/machinesubnet/openstack-provision-machinesubnet-ref.yaml [hunk 5]
```diff
@@ -24,6 +27,13 @@ ref:
       default: "172.16.0.20"
     - name: BASTION_SUBNET_RANGE
       default: "10.0.0.0/16"
+      documentation: |-
+        For 'proxy' CONFIG_TYPE, BASTION_SUBNET_RANGE will be the Network CIDR, 
+        which connects the Bastion instance with the Bastion router. 
+    - name: MACHINES_SUBNET_v4_RANGE
+      default: "192.168.25.0/24"
+      documentation: |-
+        CIDR of the IPv4 Machines Subnet for 'dualstack' CONFIG_TYPE.
```
Comment (mandre) [inline ci-operator/step-registry/openstack/provision/machinesubnet/openstack-provision-machinesubnet-ref.yaml:36]: If we need a different subnet range for the dualstack job, we should pass it as a variable via the job definition like we do for example [here](https://github.com/openshift/release/blob/62162de/ci-operator/config/shiftstack/ci/shiftstack-ci-release-4.17.yaml#L72-L75).

If we need a different machine subnet because of the cloud where the job runs, we should instead consider make it part of [openstack-conf-resourcenames-commands.sh](https://github.com/openshift/release/blob/62162de2074d035e7b96ea3c1abeda466a473b17/ci-operator/step-registry/openstack/conf/resourcenames/openstack-conf-resourcenames-commands.sh).

File: ci-operator/step-registry/openstack/provision/securitygroups/openstack-provision-securitygroups-commands.sh [hunk 6]
```diff
@@ -29,7 +32,11 @@ if [[ -n "$ADDITIONAL_SECURITY_GROUP_RULES" ]]; then
 			case $service in
 				netperf)
 					echo "Adding ${service} rule to security group ${sg_id}" 
-					openstack security group rule create "$sg_id" --protocol tcp --dst-port 12865:12865 --remote-ip 0.0.0.0/0
+					openstack security group rule create "$sg_id" --protocol tcp --dst-port 12865:12865 --remote-ip 0.0.0.0/0 --description netperf
+					openstack security group rule create "$sg_id" --protocol tcp --dst-port 22865:22865 --remote-ip 0.0.0.0/0 --description iperf3
+					openstack security group rule create "$sg_id" --protocol tcp --dst-port 30000:30000 --remote-ip 0.0.0.0/0 --description uperf
```
Comment (mandre) [inline ci-operator/step-registry/openstack/provision/securitygroups/openstack-provision-securitygroups-commands.sh:37]: Can we also consider restricting the remote IP for these different rules?
